controllers/registry.go

```diff
@@ -344,45 +394,50 @@ func getRegistryImage(dpa *oadpv1alpha1.DataProtectionApplication) string {
 	return fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_REGISTRY_REPO"), os.Getenv("VELERO_REGISTRY_TAG"))
 }
 
-func (r *DPAReconciler) buildRegistryContainer(bsl *velerov1.BackupStorageLocation, dpa *oadpv1alpha1.DataProtectionApplication) ([]corev1.Container, error) {
+func (r *DPAReconciler) buildRegistryContainer(bsl *velerov1.BackupStorageLocation, dpa *oadpv1alpha1.DataProtectionApplication, containerPort *int32) ([]corev1.Container, error) {
+	if containerPort == nil {
+		containerPort = pointer.Int32(5000)
+	}
+	if *containerPort < 5000 || *containerPort > 65534 {
+		return nil, fmt.Errorf("invalid container port %v. has to be 5000 < x < 65535", containerPort)
+	}
+	debugPort := *containerPort + 1
 	envVars, err := r.getRegistryEnvVars(bsl)
 	if err != nil {
 		r.Log.Info(fmt.Sprintf("Error building registry container for backupstoragelocation %s/%s, could not fetch registry env vars", bsl.Namespace, bsl.Name))
 		return nil, err
 	}
+	// append port config to env vars
+	envVars = append(envVars, corev1.EnvVar{
+		Name: "REGISTRY_HTTP_ADDR",
+		Value: fmt.Sprintf("localhost:%v", *containerPort),
+	}, corev1.EnvVar{
+		Name: "REGISTRY_HTTP_DEBUG_ADDR",
+		Value: fmt.Sprintf("localhost:%v", debugPort),
+	})
+	probe := corev1.Probe{
+				Handler: corev1.Handler{
+					Exec: &corev1.ExecAction{
+						Command: []string{"/bin/sh", "-c", "curl --fail localhost:" + fmt.Sprintf("%v",*containerPort) +"/v2/_catalog?n=5"},
+					},
+				},
+				PeriodSeconds:       5,
+				TimeoutSeconds:      3,
+				InitialDelaySeconds: 15,
+			}
 	containers := []corev1.Container{
 		{
 			Image: getRegistryImage(dpa),
 			Name:  registryName(bsl) + "-container",
 			Ports: []corev1.ContainerPort{
 				{
-					ContainerPort: 5000,
+					ContainerPort: *containerPort,
```

Comment:
do we need to set the port here? from the slack message seemed like this was interpod communication.

Reply:
Looks like it doesn't do anything. Just informational. Removing caused no issues.
> ports 
array
List of ports to expose from the container. Exposing a port here gives the system additional information about the network connections a container uses, but is primarily informational. Not specifying a port here DOES NOT prevent that port from being exposed. Any port which is listening on the default "0.0.0.0" address inside a container will be accessible from the network. Cannot be updated.